### ISSUE TITLE
[scan] derivedDataPath getting added to xcodebuild command twice, causing error

### ISSUE THREAD

**Issue by TheMetalCode** [fastlane/fastlane#16558]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Once I updated to the latest release (`2.149.0`), I have noticed that `fastlane scan` is appending `-derivedDataPath` to the `xcodebuild` command twice, which causes it to fail.

This is the lane I use in my Fastfile to run scan:

```ruby
  lane :build_for_ci do
    scan(
      clean: false,
      scheme: $scheme,
      workspace: $workspace,
      configuration: "UnitTest",
      build_for_testing: true,
      buildlog_path: "build_logs",
      skip_build: true,
      devices: $devices
    )
  end
```
This only appears to be affecting `scan` - my `gym` actions are operating as expected.

##### Command executed

`bundle exec fastlane build_for_ci` (see Issue Description, this is a custom lane in my Fastfile that wraps the `scan` action)

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
    +--------------------------------------------+---------+-------------------------------------------------+
|                                              [32mUsed plugins[0m                                              |
+--------------------------------------------+---------+-------------------------------------------------+
| Plugin                                     | Version | Action                                          |
+--------------------------------------------+---------+-------------------------------------------------+
| fastlane-plugin-update_project_codesigning | 0.2.0   | update_project_codesigning                      |
| fastlane-plugin-trainer                    | 0.4.1   | trainer                                         |
| fastlane-plugin-firebase_app_distribution  | 0.1.4   | firebase_app_distribution                       |
| fastlane-plugin-bugsnag                    | 1.4.2   | upload_symbols_to_bugsnag send_build_to_bugsnag |
+--------------------------------------------+---------+-------------------------------------------------+

[12:00:28]: [32m----------------------------------------[0m
[12:00:28]: [32m--- Step: Verifying fastlane version ---[0m
[12:00:28]: [32m----------------------------------------[0m
[12:00:28]: Your fastlane version 2.149.0 matches the minimum requirement of 2.121.1  ✅
[12:00:28]: [32m------------------------------[0m
[12:00:28]: [32m--- Step: default_platform ---[0m
[12:00:28]: [32m------------------------------[0m
[12:00:28]: [32mDriving the lane 'ios build_for_ci' 🚀[0m
[12:00:29]: [32m------------------[0m
[12:00:29]: [32m--- Step: scan ---[0m
[12:00:29]: [32m------------------[0m
[12:00:29]: [33mResolving Swift Package Manager dependencies...[0m
[12:00:29]: [36m$ xcodebuild -resolvePackageDependencies -workspace [REDACTED].xcworkspace -scheme [REDACTED] -configuration UnitTest[0m
[12:00:29]: ▸ [35mCommand line invocation:[0m
[12:00:29]: ▸ [35m    /Applications/Xcode11.4.1.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -workspace [REDACTED].xcworkspace -scheme [REDACTED] -configuration UnitTest[0m
[12:00:31]: ▸ [35mresolved source packages: [0m
[12:00:31]: [36m$ xcodebuild -showBuildSettings -workspace [REDACTED].xcworkspace -scheme [REDACTED] -configuration UnitTest[0m
[12:00:34]: [33mCommand timed out after 3 seconds on try 1 of 4, trying again with a 6 second timeout...[0m

+------------------------------------------------+---------------------------------------------------------------------------------------------+
|                                                           [32mSummary for scan 2.149.0[0m                                                           |
+------------------------------------------------+---------------------------------------------------------------------------------------------+
| clean                                          | false                                                                                       |
| scheme                                         | [REDACTED]                                                                                   |
| workspace                                      | [REDACTED].xcworkspace                                                                       |
| configuration                                  | UnitTest                                                                                    |
| build_for_testing                              | true                                                                                        |
| buildlog_path                                  | build_logs                                                                                  |
| skip_build                                     | true                                                                                        |
| devices                                        | ["iPhone 11 Pro Max (13.4.1)"]                                                              |
| derived_data_path                              | /Users/jhagglund/Library/Developer/Xcode/DerivedData/[REDACTED]-gfbwubeiqylhpkgygoaujpvefsux |
| skip_detect_devices                            | false                                                                                       |
| force_quit_simulator                           | false                                                                                       |
| reset_simulator                                | false                                                                                       |
| disable_slide_to_type                          | true                                                                                        |
| reinstall_app                                  | false                                                                                       |
| app_identifier                                 | [REDACTED]                                                                   |
| open_report                                    | false                                                                                       |
| output_directory                               | ./fastlane/test_output                                                                      |
| output_types                                   | html,junit                                                                                  |
| include_simulator_logs                         | false                                                                                       |
| should_zip_build_products                      | false                                                                                       |
| result_bundle                                  | false                                                                                       |
| use_clang_report_name                          | false                                                                                       |
| disable_concurrent_testing                     | false                                                                                       |
| slack_use_webhook_configured_username_and_icon | false                                                                                       |
| slack_username                                 | fastlane                                                                                    |
| slack_icon_url                                 | https://fastlane.tools/assets/img/fastlane_icon.png                                         |
| skip_slack                                     | false                                                                                       |
| slack_only_on_failure                          | false                                                                                       |
| xcodebuild_command                             | env NSUnbufferedIO=YES xcodebuild                                                           |
| fail_build                                     | true                                                                                        |
| xcode_path                                     | /Applications/Xcode11.4.1.app                                                               |
+------------------------------------------------+---------------------------------------------------------------------------------------------+

[12:00:40]: Disabling 'Slide to Type' iPhone 11 Pro Max
[12:00:40]: [36m$ /usr/libexec/PlistBuddy -c "Add :KeyboardContinuousPathEnabled bool false" /Users/jhagglund/Library/Developer/CoreSimulator/Devices/3B0AAC9C-C5C6-4C86-9282-2C8B965D35A3/data/Library/Preferences/com.apple.keyboard.ContinuousPath.plist >/dev/null 2>&1[0m
[12:00:40]: [36m$ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -workspace [REDACTED].xcworkspace -scheme [REDACTED] -configuration UnitTest -derivedDataPath /Users/jhagglund/Library/Developer/Xcode/DerivedData/[REDACTED]-gfbwubeiqylhpkgygoaujpvefsux -destination 'platform=iOS Simulator,id=3B0AAC9C-C5C6-4C86-9282-2C8B965D35A3' -derivedDataPath '/Users/jhagglund/Library/Developer/Xcode/DerivedData/[REDACTED]-gfbwubeiqylhpkgygoaujpvefsux' build-for-testing | tee '/Users/jhagglund/dev/[REDACTED]/build_logs/[REDACTED]-[REDACTED].log' | xcpretty  --report html --output '/Users/jhagglund/dev/[REDACTED]/fastlane/test_output/report.html' --report junit --output '/Users/jhagglund/dev/[REDACTED]/fastlane/test_output/report.junit' --report junit --output '/var/folders/c1/1nyddphd64x2jbzx1mgv7wlw0000gn/T/junit_report20200603-12600-1ah5yjr' [0m
[12:00:40]: ▸ [35mLoading...[0m
[12:00:40]: ▸ [35mxcodebuild: error: option '-derivedDataPath' may only be provided once[0m
[12:00:40]: ▸ [35mUsage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings [-json]] [<buildsetting>=<value>]... [<buildaction>]...[0m
[12:00:40]: ▸ [35m       xcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings [-json]] [-showdestinations] [<buildsetting>=<value>]... [<buildaction>]...[0m
[12:00:40]: ▸ [35m       xcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [-showdestinations] [<buildsetting>=<value>]... [<buildaction>]...[0m
[12:00:40]: ▸ [35m       xcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [-json] [<infoitem>] ][0m
[12:00:40]: ▸ [35m       xcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]] [-json][0m
[12:00:40]: ▸ [35m       xcodebuild -showsdks [-json][0m
[12:00:40]: ▸ [35m       xcodebuild -exportArchive -archivePath <xcarchivepath> [-exportPath <destinationpath>] -exportOptionsPlist <plistpath>[0m
[12:00:40]: ▸ [35m       xcodebuild -exportNotarizedApp -archivePath <xcarchivepath> -exportPath <destinationpath>[0m
[12:00:40]: ▸ [35m       xcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...[-includeScreenshots]][0m
[12:00:40]: ▸ [35m       xcodebuild -importLocalizations -localizationPath <path> -project <projectname>[0m
[12:00:40]: ▸ [35m       xcodebuild -resolvePackageDependencies [-project <projectname>|-workspace <workspacename>] -clonedSourcePackagesDirPath <path>[0m
[12:00:40]: ▸ [35m       xcodebuild -create-xcframework [-help] [-framework <path>] [-library <path> [-headers <path>]] -output <path>[0m
[12:00:40]: ▸ [35mOptions:[0m
[12:00:40]: ▸ [35m    -usage                                                   print brief usage[0m
[12:00:40]: ▸ [35m    -help                                                    print complete usage[0m
[12:00:40]: ▸ [35m    -verbose                                                 provide additional status output[0m
[12:00:40]: ▸ [35m    -license                                                 show the Xcode and SDK license agreements[0m
[12:00:40]: ▸ [35m    -checkFirstLaunchStatus                                  Check if any First Launch tasks need to be performed[0m
[12:00:40]: ▸ [35m    -runFirstLaunch                                          install packages and agree to the license[0m
[12:00:40]: ▸ [35m    -project NAME                                            build the project NAME[0m
[12:00:40]: ▸ [35m    -target NAME                                             build the target NAME[0m
[12:00:40]: ▸ [35m    -alltargets                                              build all targets[0m
[12:00:40]: ▸ [35m    -workspace NAME                                          build the workspace NAME[0m
[12:00:40]: ▸ [35m    -scheme NAME                                             build the scheme NAME[0m
[12:00:40]: ▸ [35m    -configuration NAME                                      use the build configuration NAME for building each target[0m
[12:00:40]: ▸ [35m    -xcconfig PATH                                           apply the build settings defined in the file at PATH as overrides[0m
[12:00:40]: ▸ [35m    -arch ARCH                                               build each target for the architecture ARCH; this will override architectures defined in the project[0m
[12:00:40]: ▸ [35m    -sdk SDK                                                 use SDK as the name or path of the base SDK when building the project[0m
[12:00:40]: ▸ [35m    -toolchain NAME                                          use the toolchain with identifier or name NAME[0m
[12:00:40]: ▸ [35m    -destination DESTINATIONSPECIFIER                        use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)[0m
[12:00:40]: ▸ [35m    -destination-timeout TIMEOUT                             wait for TIMEOUT seconds while searching for the destination device[0m
[12:00:40]: ▸ [35m    -parallelizeTargets                                      build independent targets in parallel[0m
[12:00:40]: ▸ [35m    -jobs NUMBER                                             specify the maximum number of concurrent build operations[0m
[12:00:40]: ▸ [35m    -maximum-concurrent-test-device-destinations NUMBER      the maximum number of device destinations to test on concurrently[0m
[12:00:40]: ▸ [35m    -maximum-concurrent-test-simulator-destinations NUMBER   the maximum number of simulator destinations to test on concurrently[0m
[12:00:40]: ▸ [35m    -parallel-testing-enabled YES|NO                         overrides the per-target setting in the scheme[0m
[12:00:40]: ▸ [35m    -parallel-testing-worker-count NUMBER                    the exact number of test runners that will be spawned during parallel testing[0m
[12:00:40]: ▸ [35m    -maximum-parallel-testing-workers NUMBER                 the maximum number of test runners that will be spawned during parallel testing[0m
[12:00:40]: ▸ [35m    -dry-run                                                 do everything except actually running the commands[0m
[12:00:40]: ▸ [35m    -quiet                                                   do not print any output except for warnings and errors[0m
[12:00:40]: ▸ [35m    -hideShellScriptEnvironment                              don't show shell script environment variables in build log[0m
[12:00:40]: ▸ [35m    -showsdks                                                display a compact list of the installed SDKs[0m
[12:00:40]: ▸ [35m    -showdestinations                                        display a list of destinations[0m
[12:00:40]: ▸ [35m    -showTestPlans                                           display a list of test plans[0m
[12:00:40]: ▸ [35m    -showBuildSettings                                       display a list of build settings and values[0m
[12:00:40]: ▸ [35m    -showBuildSettingsForIndex                               display build settings for the index service[0m
[12:00:40]: ▸ [35m    -list                                                    lists the targets and configurations in a project, or the schemes in a workspace[0m
[12:00:40]: ▸ [35m    -find-executable NAME                                    display the full path to executable NAME in the provided SDK and toolchain[0m
[12:00:40]: ▸ [35m    -find-library NAME                                       display the full path to library NAME in the provided SDK and toolchain[0m
[12:00:40]: ▸ [35m    -version                                                 display the version of Xcode; with -sdk will display info about one or all installed SDKs[0m
[12:00:40]: ▸ [35m    -enableAddressSanitizer YES|NO                           turn the address sanitizer on or off[0m
[12:00:40]: ▸ [35m    -enableThreadSanitizer YES|NO                            turn the thread sanitizer on or off[0m
[12:00:40]: ▸ [35m    -enableUndefinedBehaviorSanitizer YES|NO                 turn the undefined behavior sanitizer on or off[0m
[12:00:40]: ▸ [35m    -resultBundlePath PATH                                   specifies the directory where a result bundle describing what occurred will be placed[0m
[12:00:40]: ▸ [35m    -resultStreamPath PATH                                   specifies the file where a result stream will be written to (the file must already exist)[0m
[12:00:40]: ▸ [35m    -resultBundleVersion 3 [default]                         specifies which result bundle version should be used[0m
[12:00:40]: ▸ [35m    -clonedSourcePackagesDirPath PATH                        specifies the directory to which remote source packages are fetch or expected to be found[0m
[12:00:40]: ▸ [35m    -derivedDataPath PATH                                    specifies the directory where build products and other derived data will go[0m
[12:00:40]: ▸ [35m    -archivePath PATH                                        specifies the directory where any created archives will be placed, or the archive that should be exported[0m
[12:00:40]: ▸ [35m    -exportArchive                                           specifies that an archive should be exported[0m
[12:00:40]: ▸ [35m    -exportNotarizedApp                                      export an archive that has been notarized by Apple[0m
[12:00:40]: ▸ [35m    -exportOptionsPlist PATH                                 specifies a path to a plist file that configures archive exporting[0m
[12:00:40]: ▸ [35m    -enableCodeCoverage YES|NO                               turn code coverage on or off when testing[0m
[12:00:40]: ▸ [35m    -exportPath PATH                                         specifies the destination for the product exported from an archive[0m
[12:00:40]: ▸ [35m    -skipUnavailableActions                                  specifies that scheme actions that cannot be performed should be skipped instead of causing a failure[0m
[12:00:40]: ▸ [35m    -exportLocalizations                                     exports completed and outstanding project localizations[0m
[12:00:40]: ▸ [35m    -importLocalizations                                     imports localizations for a project, assuming any necessary localized resources have been created in Xcode[0m
[12:00:40]: ▸ [35m    -localizationPath                                        specifies a path to XLIFF localization files[0m
[12:00:40]: ▸ [35m    -exportLanguage                                          specifies multiple optional ISO 639-1 languages included in a localization export[0m
[12:00:40]: ▸ [35m    -xcroot                                                  specifies a path to a .xcroot to use for building and/or testing[0m
[12:00:40]: ▸ [35m    -xctestrun                                               specifies a path to a test run specification[0m
[12:00:40]: ▸ [35m    -testPlan                                                specifies the name of the test plan associated with the scheme to use for testing[0m
[12:00:40]: ▸ [35m    -only-testing                                            constrains testing by specifying tests to include, and excluding other tests[0m
[12:00:40]: ▸ [35m    -only-testing:TEST-IDENTIFIER                            constrains testing by specifying tests to include, and excluding other tests[0m
[12:00:40]: ▸ [35m    -skip-testing                                            constrains testing by specifying tests to exclude, but including other tests[0m
[12:00:40]: ▸ [35m    -skip-testing:TEST-IDENTIFIER                            constrains testing by specifying tests to exclude, but including other tests[0m
[12:00:40]: ▸ [35m    -test-timeouts-enabled YES|NO                            enable or disable test timeout behavior[0m
[12:00:40]: ▸ [35m    -default-test-execution-time-allowance                   the default execution time an individual test is given to execute, if test timeouts are enabled[0m
[12:00:40]: ▸ [35m    -maximum-test-execution-time-allowance                   the maximum execution time an individual test is given to execute, regardless of the test's preferred allowance[0m
[12:00:40]: ▸ [35m    -only-test-configuration                                 constrains testing by specifying test configurations to include, and excluding other test configurations[0m
[12:00:40]: ▸ [35m    -skip-test-configuration                                 constrains testing by specifying test configurations to exclude, but including other test configurations[0m
[12:00:40]: ▸ [35m    -testLanguage                                            constrains testing by specifying ISO 639-1 language in which to run the tests[0m
[12:00:40]: ▸ [35m    -testRegion                                              constrains testing by specifying ISO 3166-1 region in which to run the tests[0m
[12:00:40]: ▸ [35m    -resolvePackageDependencies                              resolves any Swift package dependencies referenced by the project or workspace[0m
[12:00:40]: ▸ [35m    -disableAutomaticPackageResolution                       prevents packages from automatically being resolved to versions other than those recorded in the `Package.resolved` file[0m
[12:00:40]: ▸ [35m    -json                                                    output as JSON (note: -json implies -quiet)[0m
[12:00:40]: ▸ [35m    -allowProvisioningUpdates                                Allow xcodebuild to communicate with the Apple Developer website. For automatically signed targets, xcodebuild will create and update profiles, app IDs, and certificates. For manually signed targets, xcodebuild will download missing or updated provisioning profiles. Requires a developer account to have been added in Xcode's Accounts preference pane.[0m
[12:00:40]: ▸ [35m    -allowProvisioningDeviceRegistration                     Allow xcodebuild to register your destination device on the developer portal if necessary. This flag only takes effect if -allowProvisioningUpdates is also passed.[0m
[12:00:40]: ▸ [35m    -showBuildTimingSummary                                  display a report of the timings of all the commands invoked during the build[0m
[12:00:40]: ▸ [35m    -create-xcframework                                      create an xcframework from prebuilt libraries; -help for more information.[0m
xcodebuild: error: option '-derivedDataPath' may only be provided once

Usage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings [-json]] [<buildsetting>=<value>]... [<buildaction>]...
       xcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings [-json]] [-showdestinations] [<buildsetting>=<value>]... [<buildaction>]...
       xcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [-showdestinations] [<buildsetting>=<value>]... [<buildaction>]...
       xcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [-json] [<infoitem>] ]
       xcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]] [-json]
       xcodebuild -showsdks [-json]
       xcodebuild -exportArchive -archivePath <xcarchivepath> [-exportPath <destinationpath>] -exportOptionsPlist <plistpath>
       xcodebuild -exportNotarizedApp -archivePath <xcarchivepath> -exportPath <destinationpath>
       xcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...[-includeScreenshots]]
       xcodebuild -importLocalizations -localizationPath <path> -project <projectname>
       xcodebuild -resolvePackageDependencies [-project <projectname>|-workspace <workspacename>] -clonedSourcePackagesDirPath <path>
       xcodebuild -create-xcframework [-help] [-framework <path>] [-library <path> [-headers <path>]] -output <path>

Options:
    -usage                                                   print brief usage
    -help                                                    print complete usage
    -verbose                                                 provide additional status output
    -license                                                 show the Xcode and SDK license agreements
    -checkFirstLaunchStatus                                  Check if any First Launch tasks need to be performed
    -runFirstLaunch                                          install packages and agree to the license
    -project NAME                                            build the project NAME
    -target NAME                                             build the target NAME
    -alltargets                                              build all targets
    -workspace NAME                                          build the workspace NAME
    -scheme NAME                                             build the scheme NAME
    -configuration NAME                                      use the build configuration NAME for building each target
    -xcconfig PATH                                           apply the build settings defined in the file at PATH as overrides
    -arch ARCH                                               build each target for the architecture ARCH; this will override architectures defined in the project
    -sdk SDK                                                 use SDK as the name or path of the base SDK when building the project
    -toolchain NAME                                          use the toolchain with identifier or name NAME
    -destination DESTINATIONSPECIFIER                        use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)
    -destination-timeout TIMEOUT                             wait for TIMEOUT seconds while searching for the destination device
    -parallelizeTargets                                      build independent targets in parallel
    -jobs NUMBER                                             specify the maximum number of concurrent build operations
    -maximum-concurrent-test-device-destinations NUMBER      the maximum number of device destinations to test on concurrently
    -maximum-concurrent-test-simulator-destinations NUMBER   the maximum number of simulator destinations to test on concurrently
    -parallel-testing-enabled YES|NO                         overrides the per-target setting in the scheme
    -parallel-testing-worker-count NUMBER                    the exact number of test runners that will be spawned during parallel testing
    -maximum-parallel-testing-workers NUMBER                 the maximum number of test runners that will be spawned during parallel testing
    -dry-run                                                 do everything except actually running the commands
    -quiet                                                   do not print any output except for warnings and errors
    -hideShellScriptEnvironment                              don't show shell script environment variables in build log
    -showsdks                                                display a compact list of the installed SDKs
    -showdestinations                                        display a list of destinations
    -showTestPlans                                           display a list of test plans
    -showBuildSettings                                       display a list of build settings and values
    -showBuildSettingsForIndex                               display build settings for the index service
    -list                                                    lists the targets and configurations in a project, or the schemes in a workspace
    -find-executable NAME                                    display the full path to executable NAME in the provided SDK and toolchain
    -find-library NAME                                       display the full path to library NAME in the provided SDK and toolchain
    -version                                                 display the version of Xcode; with -sdk will display info about one or all installed SDKs
    -enableAddressSanitizer YES|NO                           turn the address sanitizer on or off
    -enableThreadSanitizer YES|NO                            turn the thread sanitizer on or off
    -enableUndefinedBehaviorSanitizer YES|NO                 turn the undefined behavior sanitizer on or off
    -resultBundlePath PATH                                   specifies the directory where a result bundle describing what occurred will be placed
    -resultStreamPath PATH                                   specifies the file where a result stream will be written to (the file must already exist)
    -resultBundleVersion 3 [default]                         specifies which result bundle version should be used
    -clonedSourcePackagesDirPath PATH                        specifies the directory to which remote source packages are fetch or expected to be found
    -derivedDataPath PATH                                    specifies the directory where build products and other derived data will go
    -archivePath PATH                                        specifies the directory where any created archives will be placed, or the archive that should be exported
    -exportArchive                                           specifies that an archive should be exported
    -exportNotarizedApp                                      export an archive that has been notarized by Apple
    -exportOptionsPlist PATH                                 specifies a path to a plist file that configures archive exporting
    -enableCodeCoverage YES|NO                               turn code coverage on or off when testing
    -exportPath PATH                                         specifies the destination for the product exported from an archive
    -skipUnavailableActions                                  specifies that scheme actions that cannot be performed should be skipped instead of causing a failure
    -exportLocalizations                                     exports completed and outstanding project localizations
    -importLocalizations                                     imports localizations for a project, assuming any necessary localized resources have been created in Xcode
    -localizationPath                                        specifies a path to XLIFF localization files
    -exportLanguage                                          specifies multiple optional ISO 639-1 languages included in a localization export
    -xcroot                                                  specifies a path to a .xcroot to use for building and/or testing
    -xctestrun                                               specifies a path to a test run specification
    -testPlan                                                specifies the name of the test plan associated with the scheme to use for testing
    -only-testing                                            constrains testing by specifying tests to include, and excluding other tests
    -only-testing:TEST-IDENTIFIER                            constrains testing by specifying tests to include, and excluding other tests
    -skip-testing                                            constrains testing by specifying tests to exclude, but including other tests
    -skip-testing:TEST-IDENTIFIER                            constrains testing by specifying tests to exclude, but including other tests
    -test-timeouts-enabled YES|NO                            enable or disable test timeout behavior
    -default-test-execution-time-allowance                   the default execution time an individual test is given to execute, if test timeouts are enabled
    -maximum-test-execution-time-allowance                   the maximum execution time an individual test is given to execute, regardless of the test's preferred allowance
    -only-test-configuration                                 constrains testing by specifying test configurations to include, and excluding other test configurations
    -skip-test-configuration                                 constrains testing by specifying test configurations to exclude, but including other test configurations
    -testLanguage                                            constrains testing by specifying ISO 639-1 language in which to run the tests
    -testRegion                                              constrains testing by specifying ISO 3166-1 region in which to run the tests
    -resolvePackageDependencies                              resolves any Swift package dependencies referenced by the project or workspace
    -disableAutomaticPackageResolution                       prevents packages from automatically being resolved to versions other than those recorded in the `Package.resolved` file
    -json                                                    output as JSON (note: -json implies -quiet)
    -allowProvisioningUpdates                                Allow xcodebuild to communicate with the Apple Developer website. For automatically signed targets, xcodebuild will create and update profiles, app IDs, and certificates. For manually signed targets, xcodebuild will download missing or updated provisioning profiles. Requires a developer account to have been added in Xcode's Accounts preference pane.
    -allowProvisioningDeviceRegistration                     Allow xcodebuild to register your destination device on the developer portal if necessary. This flag only takes effect if -allowProvisioningUpdates is also passed.
    -showBuildTimingSummary                                  display a report of the timings of all the commands invoked during the build
    -create-xcframework                                      create an xcframework from prebuilt libraries; -help for more information.
[12:00:41]: [31mExit status: 64[0m
+----------------------------+---------------------------------------------------------------------------------------------+
|                                                       [33mLane Context[0m                                                       |
+----------------------------+---------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM           | ios                                                                                         |
| PLATFORM_NAME              | ios                                                                                         |
| LANE_NAME                  | ios build_for_ci                                                                            |
| SCAN_DERIVED_DATA_PATH     | /Users/jhagglund/Library/Developer/Xcode/DerivedData/[REDACTED]-gfbwubeiqylhpkgygoaujpvefsux |
| SCAN_GENERATED_PLIST_FILES | []                                                                                          |
| SCAN_GENERATED_PLIST_FILE  |                                                                                             |
+----------------------------+---------------------------------------------------------------------------------------------+
[12:00:41]: [31mError building/testing the application. See the log above.[0m

+------+----------------------------+-------------+
|                [32mfastlane summary[0m                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 💥   | [31mscan[0m                       | 12          |
+------+----------------------------+-------------+

[12:00:41]: [31mfastlane finished with errors[0m
  </pre>
</details>

### Environment

<details>
  <pre> <details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                             |
| --------------------------- | ------------------------------------------------- |
| OS                          | 10.15.5                                           |
| Ruby                        | 2.6.6                                             |
| Bundler?                    | true                                              |
| Git                         | git version 2.27.0                                |
| Installation Source         | ~/.rbenv/versions/2.6.6/bin/fastlane              |
| Host                        | Mac OS X 10.15.5 (19F101)                         |
| Ruby Lib Dir                | ~/.rbenv/versions/2.6.6/lib                       |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                       |
| Is contained                | false                                             |
| Is homebrew                 | false                                             |
| Is installed via Fabric.app | false                                             |
| Xcode Path                  | /Applications/Xcode11.4.1.app/Contents/Developer/ |
| Xcode Version               | 11.4.1                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.121.1"

default_platform :ios

[REDACTED]

platform :ios do
  before_all do
    if ENV['CI']
      # Necessary because CircleCI has multiple XCode versions
      xcode_select "/Applications/Xcode-11.4.1.app"
    end
  end

  [REDACTED]

  desc "Performs a build for CI setup purposes, using the UnitTest configuration"
  lane :build_for_ci do
    scan(
      clean: false,
      scheme: $scheme,
      workspace: $workspace,
      configuration: "UnitTest",
      build_for_testing: true,
      buildlog_path: "build_logs",
      skip_build: true,
      devices: $devices
    )
  end

  [REDACTED]

  after_all do |lane|
    # This block is called, only if the executed lane was successful
    if ENV["CI"]
      slack(
        message: [REDACTED],
        success: true,
        fail_on_error: false,
        default_payloads: [:lane, :test_result, :git_branch, :git_author]
      )
    end
  end

  error do |lane, exception|
    if ENV["CI"]
      slack(
        message: exception.message,
        success: false,
        fail_on_error: false
      )
    end
  end
end

# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier [REDACTED] # The bundle identifier of your app

team_id [REDACTED]  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.149.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                     | Version | Update-Status |
| ------------------------------------------ | ------- | ------------- |
| fastlane-plugin-update_project_codesigning | 0.2.0   | ✅ Up-To-Date  |
| fastlane-plugin-trainer                    | 0.4.1   | ✅ Up-To-Date  |
| fastlane-plugin-firebase_app_distribution  | 0.1.4   | ✅ Up-To-Date  |
| fastlane-plugin-bugsnag                    | 1.4.2   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                        | Version      |
| ------------------------------------------ | ------------ |
| did_you_mean                               | 1.3.0        |
| bundler                                    | 2.1.4        |
| CFPropertyList                             | 3.0.2        |
| ZenTest                                    | 4.12.0       |
| RubyInline                                 | 3.12.5       |
| concurrent-ruby                            | 1.1.6        |
| i18n                                       | 0.9.5        |
| minitest                                   | 5.14.1       |
| thread_safe                                | 0.3.6        |
| tzinfo                                     | 1.2.7        |
| activesupport                              | 4.2.11.3     |
| public_suffix                              | 2.0.5        |
| addressable                                | 2.7.0        |
| httpclient                                 | 2.8.3        |
| json                                       | 2.3.0        |
| algoliasearch                              | 1.27.3       |
| atomos                                     | 0.1.3        |
| aws-eventstream                            | 1.1.0        |
| aws-partitions                             | 1.323.0      |
| aws-sigv4                                  | 1.1.4        |
| jmespath                                   | 1.4.0        |
| aws-sdk-core                               | 3.97.1       |
| aws-sdk-kms                                | 1.33.0       |
| aws-sdk-s3                                 | 1.67.1       |
| babosa                                     | 1.0.3        |
| circleci                                   | 2.0.3        |
| highline                                   | 1.7.10       |
| commander-fastlane                         | 4.4.6        |
| down                                       | 5.1.1        |
| mini_portile2                              | 2.4.0        |
| nokogiri                                   | 1.10.9       |
| http-accept                                | 1.7.0        |
| unf_ext                                    | 0.0.7.7      |
| unf                                        | 0.1.4        |
| domain_name                                | 0.5.20190701 |
| http-cookie                                | 1.0.3        |
| mime-types-data                            | 3.2020.0512  |
| mime-types                                 | 3.3.1        |
| netrc                                      | 0.11.0       |
| rest-client                                | 2.1.0        |
| safe_shell                                 | 1.1.0        |
| circle-ci-utils                            | 1.7.1        |
| claide                                     | 1.0.3        |
| fuzzy_match                                | 2.0.4        |
| nap                                        | 1.1.0        |
| cocoapods-core                             | 1.8.4        |
| cocoapods-deintegrate                      | 1.0.4        |
| cocoapods-downloader                       | 1.3.0        |
| cocoapods-plugins                          | 1.0.0        |
| cocoapods-search                           | 1.0.0        |
| cocoapods-stats                            | 1.1.0        |
| cocoapods-trunk                            | 1.5.0        |
| cocoapods-try                              | 1.2.0        |
| colored2                                   | 3.1.2        |
| escape                                     | 0.0.4        |
| fourflusher                                | 2.3.1        |
| gh_inspector                               | 1.1.3        |
| molinillo                                  | 0.6.6        |
| ruby-macho                                 | 1.4.0        |
| nanaimo                                    | 0.2.6        |
| xcodeproj                                  | 1.16.0       |
| cocoapods                                  | 1.8.4        |
| rouge                                      | 2.0.7        |
| xcpretty                                   | 0.3.0        |
| cocoapods-binary                           | 0.4.4        |
| dotenv                                     | 2.7.5        |
| osx_keychain                               | 1.0.2        |
| cocoapods-keys                             | 2.1.0        |
| colored                                    | 1.2          |
| ruby-terminfo                              | 0.1.1        |
| console_table                              | 0.3.0        |
| declarative                                | 0.0.10       |
| declarative-option                         | 0.1.0        |
| descendants_tracker                        | 0.0.4        |
| digest-crc                                 | 0.5.1        |
| emoji_regex                                | 1.0.1        |
| excon                                      | 0.73.0       |
| multipart-post                             | 2.0.0        |
| faraday                                    | 0.17.3       |
| faraday-cookie_jar                         | 0.0.6        |
| faraday_middleware                         | 0.14.0       |
| fastimage                                  | 2.1.7        |
| jwt                                        | 2.1.0        |
| memoist                                    | 0.16.2       |
| multi_json                                 | 1.14.1       |
| os                                         | 1.1.0        |
| signet                                     | 0.14.0       |
| googleauth                                 | 0.12.0       |
| mini_mime                                  | 1.0.2        |
| uber                                       | 0.1.0        |
| representable                              | 3.0.4        |
| retriable                                  | 3.1.2        |
| google-api-client                          | 0.38.0       |
| google-cloud-env                           | 1.3.2        |
| google-cloud-errors                        | 1.0.1        |
| google-cloud-core                          | 1.5.0        |
| google-cloud-storage                       | 1.26.2       |
| mini_magick                                | 4.10.1       |
| multi_xml                                  | 0.6.0        |
| plist                                      | 3.5.0        |
| rubyzip                                    | 1.3.0        |
| security                                   | 0.1.3        |
| naturally                                  | 2.2.0        |
| simctl                                     | 1.6.8        |
| slack-notifier                             | 2.3.2        |
| terminal-notifier                          | 2.0.0        |
| unicode-display_width                      | 1.7.0        |
| terminal-table                             | 1.8.0        |
| tty-screen                                 | 0.8.0        |
| tty-cursor                                 | 0.7.1        |
| tty-spinner                                | 0.9.3        |
| word_wrap                                  | 1.0.0        |
| xcpretty-travis-formatter                  | 1.0.0        |
| rchardet                                   | 1.8.0        |
| git                                        | 1.7.0        |
| xml-simple                                 | 1.1.5        |
| fastlane-plugin-bugsnag                    | 1.4.2        |
| fastlane-plugin-firebase_app_distribution  | 0.1.4        |
| trainer                                    | 0.9.1        |
| fastlane-plugin-trainer                    | 0.4.1        |
| fastlane-plugin-update_project_codesigning | 0.2.0        |
| ffi                                        | 1.13.0       |
| frontkick                                  | 0.5.7        |
| hashie                                     | 3.6.0        |
| rack                                       | 2.2.2        |
| oauth2                                     | 1.4.4        |
| github_api                                 | 0.18.2       |
| mustache                                   | 1.1.1        |
| open4                                      | 1.3.4        |
| redcarpet                                  | 3.5.0        |
| sassc                                      | 2.4.0        |
| sqlite3                                    | 1.4.2        |
| liferaft                                   | 0.0.6        |
| xcinvoke                                   | 0.3.0        |
| jazzy                                      | 0.13.4       |
| xcresult                                   | 0.2.1        |
| xcov                                       | 1.7.3        |
</details>

*generated on:* **2020-06-03**
</details> </pre>
</details>

**Comment by jcharr1** [comment]
I started having this same problem with 2.149.0, too.

**Comment by sbporter** [comment]
Same. Builds yesterday using 2.148.1 worked perfectly with only a single -derivedDataPath being set. Today it is being added twice on 2.149.0

**Comment by kylekurz** [comment]
+1

**Comment by fatihyildizhan** [comment]
+1

**Comment by cairodrigo** [comment]
+1

**Comment by jhandguy** [comment]
Also experiencing this issue, a recently merged and released PR that seems related: https://github.com/fastlane/fastlane/pull/16534

**Comment by TheMetalCode** [comment]
@jhandguy That's my suspicion too but it had a unit test that past with a strict equality assertion (meaning in that case it was just the one `-derivedData` flag). But stands to reason we're missing a test somewhere. I planned to dig in sometime today unless someone beats me to it.

**Comment by jhandguy** [comment]
This unit test only tests `project.rb`, my guess is that the error only occurs with `scan` which most probably uses `project.rb` under the hood.

**Comment by selin194** [comment]
+1

**Comment by elindberg-kt** [comment]
+1

**Comment by Shadester** [comment]
+1

**Comment by ButkiewiczP** [comment]
Can confirm this is happening with (at least) `scan`. Started after updating to 2.149.

**Comment by rickthefree** [comment]
+1 getting the same error in 2.149.0 in our CI builds.

**Comment by Mc231** [comment]
+1

**Comment by tundsD53** [comment]
+ 1

**Comment by kbendelow** [comment]
+1

**Comment by dderby** [comment]
This is a pretty serious regression that is breaking builds that run automated tests.  It's affecting an enormous number of users.  Can someone just revert PR #16534, put out an emergency release, and then fix it properly later?

CC @sudachi808 @janpio

**Comment by TheMetalCode** [comment]
https://github.com/fastlane/fastlane/pull/16559 illustrates where I think the problem is happening and provides a _potential_ fix. It'll be several hours before I can fully vet it, but wanted to get something up to at least kickstart a potential solution. I'll spend a bit more time on it this evening.

**Comment by dderby** [comment]
Thank you @TheMetalCode!

**Comment by joshdholtz** [comment]
Sorry about this! Will get that fix in a new build pushed out in a in an hour or so

**Comment by joshdholtz** [comment]
Going to revert #16559 and ship a new build. I think that PR needs to be looked into a little bit more and I’d like to get a fix for this out sooner rather than later 🤷‍♂️

**Comment by joshdholtz** [comment]
Things should be back to normal now when you update to 2.149.1! Sorry again about these issues. I should have been better about catching the side effects from that PR. Hope this didn’t cause you all too much trouble 😔 Will try to harder to prevent these things from happening going forward.

**Comment by dan-hart** [comment]
Thanks for fixing this, updating to 2.149.1 fixed this issue for me
